src/backend/cuda/kernel/nearest_neighbour.hpp

```diff
@@ -392,26 +399,26 @@ __global__ void select_matches(
                 s_dist[sid] = dist;
                 s_idx[sid]  = in_idx[f * nelem + i];
             }
-            __syncthreads();
         }
+    }
+    __syncthreads();
 
-        // Reduce best matches and find the best of them all
-        for (unsigned i = blockDim.y / 2; i > 0; i >>= 1) {
-            if (threadIdx.y < i) {
-                To dist = s_dist[sid + i];
-                if (dist < s_dist[sid]) {
-                    s_dist[sid] = dist;
-                    s_idx[sid]  = s_idx[sid + i];
-                }
-                __syncthreads();
+    // Reduce best matches and find the best of them all
+    for (unsigned i = blockDim.y / 2; i > 0; i >>= 1) {
+        if (threadIdx.y < i) {
+            To dist = s_dist[sid + i];
+            if (dist < s_dist[sid]) {
+                s_dist[sid] = dist;
+                s_idx[sid]  = s_idx[sid + i];
             }
+            __syncthreads();
```

Comment:
It was already like that (look at the left). I fixed it in one place but missed this. I'll fix it.